src/uct/cuda/cuda_ipc/cuda_ipc_md.c

```diff
@@ -158,24 +192,48 @@ static ucs_status_t uct_cuda_ipc_query_md_resources(uct_md_resource_desc_t **res
     return uct_single_md_resource(&uct_cuda_ipc_md_component, resources_p, num_resources_p);
 }
 
+static void uct_cuda_ipc_md_close(uct_md_h uct_md)
+{
+    uct_cuda_ipc_md_t *md = ucs_derived_of(uct_md, uct_cuda_ipc_md_t);
+    if (NULL != md->uuid_map) {
+        ucs_free(md->uuid_map);
+    }
+    ucs_free(md);
+}
+
 static ucs_status_t uct_cuda_ipc_md_open(const char *md_name, const uct_md_config_t *md_config,
                                          uct_md_h *md_p)
 {
+    uct_cuda_ipc_md_t *cuda_ipc_md;
+    ucs_status_t status;
+
+    cuda_ipc_md = ucs_malloc(sizeof(*cuda_ipc_md), "uct_cuda_ipc_md_t");
+    if (cuda_ipc_md == NULL) {
+        ucs_error("Failed to allocate memory for uct_cuda_ipc_md_t");
+        status = UCS_ERR_NO_MEMORY;
+        goto err;
+    }
+    
+    cuda_ipc_md->uuid_map = NULL;
+    cuda_ipc_md->uuid_map_len = -1;
+    
     static uct_md_ops_t md_ops = {
-        .close        = (void*)ucs_empty_function,
+        .close        = uct_cuda_ipc_md_close,
         .query        = uct_cuda_ipc_md_query,
         .mkey_pack    = uct_cuda_ipc_mkey_pack,
         .mem_reg      = uct_cuda_ipc_mem_reg,
         .mem_dereg    = uct_cuda_ipc_mem_dereg,
         .is_mem_type_owned = uct_cuda_is_mem_type_owned,
     };
-    static uct_md_t md = {
-        .ops          = &md_ops,
-        .component    = &uct_cuda_ipc_md_component
-    };
 
-    *md_p = &md;
+    cuda_ipc_md->super.ops = &md_ops;
+    cuda_ipc_md->super.component = &uct_cuda_ipc_md_component;
+
```

Comment:
@bureddy given `uct_cuda_ipc_md_component`, how do I extract `cuda_ipc_md`? I don't see such usage in other UCTs

Reply:
@yosefe can you suggest here if this is right approach?

Reply:
@bureddy @yosefe 

Going through an ugly approach for now by using `component->priv` to refer to `md` instance. Will this work? Refer
1. https://github.com/openucx/ucx/pull/3588/files#diff-80454453c5f1d6f7c47a5a3750d739d4R57
2. https://github.com/openucx/ucx/pull/3588/files#diff-80454453c5f1d6f7c47a5a3750d739d4R248

Reply:
With this I'm able to successfully use NVLINKs for the following case:
```shell
# ucx_perftest will use device 0 at server and device 2 at client which are NVLINK-connected
server:
akvenkatesh@prm-dgx-07:~/ucx-github/build$ CUDA_VISIBLE_DEVICES=0,1,2,3 ./bin/ucx_perftest -t tag_bw -b msg_pow2 -m cuda
+------------------------------------------------------------------------------------------+
| API:          protocol layer                                                             |
| Test:         tag match bandwidth                                                        |
| Data layout:  (automatic)                                                                |
| Message size: 8                                                                          |
+------------------------------------------------------------------------------------------+

client:
akvenkatesh@prm-dgx-07:~/ucx-github/build$ CUDA_VISIBLE_DEVICES=1,2,3,0 ./bin/ucx_perftest prm-dgx-07 -t tag_bw -b msg_pow2 -m cuda
+--------------+-----------------------------+---------------------+-----------------------+
|              |       latency (usec)        |   bandwidth (MB/s)  |  message rate (msg/s) |
+--------------+---------+---------+---------+----------+----------+-----------+-----------+
| # iterations | typical | average | overall |  average |  overall |   average |   overall |
+--------------+---------+---------+---------+----------+----------+-----------+-----------+
+2097152-------+---------+---------+---------+----------+----------+-----------+-----------+
           100     0.000   106.139   106.139   18843.18   18843.18        9422        9422
```

Reply:
I dont know usage of priv field here. I will let @yosefe review if using "priv" is ok